clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -661,4 +674,47 @@ class GCEUtil {
 
     return firewall
   }
+
+  /*
+    Zones that support Haswell and Ivy Bridge processors can support custom machine types up to 32 vCPUs.
+    Zones that support Sandy Bridge processors can support up to 16 vCPUs.
+  */
+  private static final def vCpuMaxByZone = [
+    'us-east1-b': 32,
+    'us-east1-c': 32,
+    'us-east1-d': 32,
+    'us-central1-a': 16,
+    'us-central1-b': 32,
+    'us-central1-c': 32,
+    'us-central1-f': 32,
+    'europe-west1-b': 16,
+    'europe-west1-c': 32,
+    'europe-west1-d': 32,
+    'asia-east1-a': 32,
+    'asia-east1-b': 32,
+    'asia-east1-c': 32
+  ]
+
+  private static String validateCustomMachineTypeName(String machineTypeName, Task task, String phase, String zone) {
+
+    def ( vCpuCount, memory ) = machineTypeName.split('-').tail().collect { it.toDouble() }
+    def memoryInGbs = memory / 1024
+
+    // Memory per vCPU must be between .9 GB and 6.5 GB
+    def maxMemory = vCpuCount * 6.5
+    def minMemory = Math.ceil((0.9 * vCpuCount) * 4) / 4
+
+    def invalidatingStatements = [
+      vCpuCount > vCpuMaxByZone[zone],
+      vCpuCount != 1 && vCpuCount % 2 == 1, // Above 1, vCPU count must be even.
+      memoryInGbs > maxMemory,
+      memoryInGbs < minMemory,
+      memory % 256 != 0 // Total memory must be a multiple of 256 MB.
+    ]
+
+    if ( invalidatingStatements.contains(true) ) {
+      updateStatusAndThrowNotFoundException("Custom machine $machineTypeName not valid.", task, phase)
```

Comment:
Could give a caller (esp. a script caller) more details as to what the validation issue is.